staging/src/k8s.io/apiserver/pkg/storage/etcd3/store.go

```diff
@@ -119,6 +119,9 @@ func (s *store) Get(ctx context.Context, key string, resourceVersion string, out
 	if err != nil {
 		return err
 	}
+	if err = s.ensureMinimumResourceVersion(resourceVersion, getResp.Header.Revision); err != nil {
```

Comment:
I'm not convinced it should be a blocker or not, but I think I need to bring up one thing - even with this change the semantic of both cache and direct etcd3 calls isn't identical.
The difference is that in watchcache, we're basing the response on resource versions of object only of that type, whereas in etcd3, based on all objects.

So imagine a situation:
- latest change of pod has RV=10
- however, there were changes to other objects and latest RV from all etcd is RV=20
- I'm sending "List pods RV=15" request:
 (a) in case of cache, I will get "RV too large"
 (b) in case of etcd, I will get the result

The potential solution would be some _frequent_ bookmarks from etcd to watchcache.
Though as I said, I'm not sure it's a blocker - this PR is a step in a good direction anyway.

@liggitt @jpbetz @lavalamp - thoughts?

Reply:
Say, hypothetically, you configured the api-server with `--etcd-servers-overrides` such that each resource type was in its own etcd. Would that result the basically the same behavior as the watch cache currently has? (If so, it seems to me that the watch cache behavior is the most strict/most correct)